### PR TITLE
Add new `Lint/UselessMethodDefinition` cop

### LINKED ISSUE #8472: New cop: `Lint/UselessMethodDefinition`
There are at least 2 cases:
1. empty constructors - this may be just overriding parent's constructor, but this is bad anyway 
2. methods just calling `super`

### PR DESCRIPTION
**Pull request by fatkodima**
Closes #8472

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -1942,6 +1942,13 @@ Lint/UselessElseWithoutRescue:
   Enabled: true
   VersionAdded: '0.17'
 
+Lint/UselessMethod:
```
Comment (bbatsov) [inline config/default.yml:1945]: I think "UselessMethodDefinition" might be a better name for this.

File: lib/rubocop/cop/lint/useless_method.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      # @example AllowComments: false
+      #   # bad
+      #   def initialize
+      #     # Comment.
+      #   end
+      #
+      class UselessMethod < Base
+        extend AutoCorrector
+
+        MSG = 'Useless method detected.'
```
Comment (bbatsov) [inline lib/rubocop/cop/lint/useless_method.rb:46]: method definition

**Comment by fatkodima** [comment]
Updated with new name.

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**